LibraryManagement.cpp

```diff
@@ -2,342 +2,332 @@
-    	cin >> firstName;
-    	cout << "Enter last name: ";
-    	cin >> lastName;
-    while(ID.length() != 13) {
-            cout<<"Enter ID: ";
-            cin>>ID;
-        if(ID.length() != 13) {
-            cout<<"ID must have 13 digits"<<endl;
         }
-	}
-
-    bookMark.users[bookMark.userCounter].setFirstName(firstName);
-    bookMark.users[bookMark.userCounter].setLastName(lastName);
-    bookMark.users[bookMark.userCounter].setID(ID);
-    bookMark.incrementUser(bookMark.userCounter);
-        cout << "\nUser added successfully."<<endl;
-        main();
-
+    }
+    else {
+        cout << "Invalid input." << endl;
+    }
+return 0;
 }
-	else {
-		cout<<"You have reached the maximum number of users that can be added!"<<endl;
-		main();
-	}
+void Bookmark::registerNewUser() {
+    string username, password;
+    cout << "Enter username: ";
+    cin >> username;
+    cout << "Enter password: ";
+    cin >> password;
+    if (this->registerUser(username, password)) {
+        cout << "You are successfully registered" << endl;
+    } else {
+        cout << "Error: username already exists." << endl;
+        exit(0);
+      }
 }
-
-void Bookmark::editUser(int userCounter) {
-    string editId, ID,firstName,lastName,choice;
-    	cout << "EDIT USER\n\n";
-    if(bookMark.userCounter==0){
-		cout<<"There are no users to edit!"<<endl;
-		main();
-	}
-		cout<<"Enter JMBG: ";
-		cin>>editId;
-
-    for (int i = 0; i < bookMark.userCounter; i++) {
-        if (bookMark.users[i].getID() == editId) {
-            cout<<"User found!\n\n";
-            cout<<"ID: "<<bookMark.users[i].getID()<<endl;
-			cout<<"First name: "<<bookMark.users[i].getFirstName()<<endl;
-			cout<<"Last name: "<<bookMark.users[i].getLastName()<<endl;
-			cout<<"\nDo you want to edit?\n[1]Yes\n[2]No\n\nEnter choice: ";
-			cin>>choice;
-			if(choice=="1"){
-				cout<<"- First name: ";
-				cin>>firstName;
-				cout<<"- Prezime: ";
-				cin>>lastName;
-			while(ID.length() != 13) {
-            	cout<<"- Enter ID: ";
-            	cin>>ID;
-
-            if(ID.length() != 13) {
-                cout<<"ID must have 13 digits"<<endl;
-			}
-			
-            bookMark.users[i].setFirstName(firstName);
-            bookMark.users[i].setLastName(lastName);
-            bookMark.users[i].setID(ID);
-                cout<<"User edited successfully!"<<endl;
-                main();
-        	}
+void Bookmark::loginUser() {
+    string username, password;
+    do {
+        cout << "Enter username: ";
+        cin >> username;
+        cout << "Enter password: ";
+        cin >> password;
+        if (this->isAlreadyRegistered(username, password)) {
+            cout << "Login successful" << endl;
+            break;
+        } else {
+            cout << "Incorrect username or password!" << endl;
+            cout << "Press 'y' to try again: ";
+            char tryAgain;
+            cin >> tryAgain;
+            tryAgain = tolower(tryAgain);
+            if (tryAgain != 'y') {
```

Comment:
`y` can also be a constant. You can declare another file which will contain only constants.